docs/bugs.json

```
@@ -1,4 +1,17 @@
 [
+    {
+        "uid": "SOL-2022-4",
+        "name": "InlineAssemblyMemorySideEffects",
+        "summary": "The Yul optimizer may incorrectly remove memory writes from inline assembly blocks, that do not access solidity variables.",
+        "description": "The Yul optimizer considers all memory writes that are never read from in the outermost Yul block as unused and removes them. This is valid for a full Yul program generated with the new via-IR code generation pipeline. However the legacy code generation path (which is still the default) also runs the Yul optimizer individually on all inline assembly blocks, which do not access local Solidity variables. Consequently, memory writes by such inline assembly blocks are removed as well, if the written memory is never read from in the same assembly block, even if the written memory is accessed later for example by a subsequent inline assembly block.",
```

I'd reword it a bit to make it clearer why asm blocks in IR are not affected. How about this?

```suggestion
        "description": "The Yul optimizer considers all memory writes in the outermost Yul block that are never read from as unused and removes them. This is valid when that Yul block is the entire Yul program, which is always the case for the Yul code generated by the new via-IR pipeline. Inline assembly blocks are never optimized in isolation when using that pipeline. However, the legacy code generation pipeline (which is still the default) runs the Yul optimizer individually on an inline assembly block if the block does not refer to any local variables defined in the surrounding Solidity code. Consequently, memory writes in such inline assembly blocks are removed as well, if the written memory is never read from in the same assembly block, even if the written memory is accessed later, for example by a subsequent inline assembly block.",
```

---

Nice, thanks :-)!